src/model/File_data.h

```diff
@@ -35,20 +35,24 @@ class File_data
             : file_meta_(file_meta),
               contents_(contents) {}
 
-    const dbo::ptr<File_meta>& file_meta() const { return file_meta_; }
-    const Bytes& contents() const { return contents_; }
+	File_data(const dbo::ptr<File_meta>& file_meta)
+			: file_meta_(file_meta) {}
 
+	const dbo::ptr<File_meta>& file_meta() const { return file_meta_; }
+	const Bytes& contents() const { return contents_; }
+	int write_and_commit();
```

Comment:
What do the `int`s returned by this function and the next two mean? If they indicate an error that should be checked then they should throw exceptions instead, probably.

Reply:
To indicate erros yep and I Will take care of that by throwing exceptions.